wherehows-web/app/components/datasets/containers/upstream-dataset.ts

```diff
@@ -146,7 +146,9 @@ export default class UpstreamDatasetContainer extends Component {
   getUpstreamMetadataTask = task(function*(
     this: UpstreamDatasetContainer
   ): IterableIterator<TaskInstance<Promise<LineageList>> | Promise<Array<IUpstreamWithComplianceMetadata>>> {
-    const upstreamLineage: LineageList = yield get(this, 'getUpstreamDatasetsTask').perform();
+    // Fallback logic, if we have lineage then use that otherwise get upstreams again
+    // prettier-ignore
+    const upstreamLineage = get(this, 'upstreamLineage') || <LineageList>(yield get(this, 'getUpstreamDatasetsTask').perform());
```

Comment:
Do you need a task modifier for `getUpstreadDatasetsTask` as well? Like `restartable` or `drop`?

Reply:
That's a good point!